### ISSUE TITLE
Create nginx/apache site

### ISSUE THREAD

**Issue by jbrooksuk** [REBELinBLUE/deployer#78]
Automatically create the nginx/Apache config file depending on the server type.

**Comment by pavankumarkatakam** [comment]
+1

**Comment by lorvent** [comment]
upon creating a project in deployer?

**Comment by djfurman** [comment]
Definitely +1

I'd love to see this become a feature where the same server can be used for a named development branch and a named staging or demo branch.

For example: 

- if I push develop (or any named development branch) `dev.project.example.com` would be accessible with this deployment for my internal team and us to play around
- if I push release/0.1.0 (or some demo version or RC named whatever) `staging.project.example.com` would be available with this release

This way a single server instance could aid with multiple projects and versions for team review, testing, internal/client demonstrations, etc.

If anybody is working on a PR for this and would like some help, send me a note, I'd like to help you out!

*On a personal note*; I just came across this project and am very impressed so far. I'm a little bogged down at the moment, but I'd be very interested in being a collaborator. I've been developing in Laravel for a little while now and have been working with Linux for as long as I've been in the technology world. Combining both in a project sounds like good times!

**Comment by emielmolenaar** [comment]
I don't think server configuration management is the job of a deployment tool. Apart from that it's kind of complex to setup this for all kinds of different projects and their requirements; making it hard to maintain and test the responsible code.

**Comment by REBELinBLUE** [comment]
Indeed, exactly why it isn't really something I've been interested in looking into
